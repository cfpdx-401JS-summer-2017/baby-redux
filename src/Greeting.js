import React from 'react';
import { connect } from 'react-redux';
import { updateSalutation, updateName, reset } from './greeting.actions';

export function Greeting({ salutation, name }) {
  return (
    <div>
      <span>{salutation}</span>
      <span>{name}</span>!
    </div>
  );
}

export default connect(state => (
  {
    salutation: state.salutation,
    name: state.name
  }
))(Greeting);

export function Editor({ dispatch }) {
  return (
    <div>
      <label>
        Salutation:
        <input type="text" onChange={({target}) => dispatch(updateSalutation(target.value))} />
      </label>
      <br/>
      <label>
        Name:
        <input type="text" onChange={({target}) => dispatch(updateName(target.value))} />
      </label>
      <br/>
      <button onClick={() => reset()} >Reset</button>
    </div>
  );
}
