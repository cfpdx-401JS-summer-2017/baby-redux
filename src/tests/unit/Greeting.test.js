import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Editor } from '../../Greeting.js';

describe('Editor', () => {

  it('shows greeting with two inputs and a button', () => {
    const wrapper = toJSON(shallow(
      <Editor salutation="Hallo" name="Freund" reset={() => {}} dispatch={() => {}} />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  // it('clears greeting when reset button is clicked', () => {
  //   const wrapper = shallow(<Editor />);
  //   expect(wrapper.find('button').simulate('click'));

  // });
  

});
