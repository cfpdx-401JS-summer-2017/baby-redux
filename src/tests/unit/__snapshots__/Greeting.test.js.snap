// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor shows greeting with two inputs and a button 1`] = `
<div>
  <div>
    <span>
      Hallo
    </span>
     
    <span>
      Freund
    </span>
    !
  </div>
  <label>
    Salutation:
    <input
      onChange={[Function]}
      type="text"
      value="Hallo"
    />
  </label>
  <br />
  <label>
    Name:
    <input
      onChange={[Function]}
      type="text"
      value="Freund"
    />
  </label>
  <br />
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;
